Fix Rocky OS compatibility

FathiQ opened the pull request:
### Proposed changes
Support Rocky OS

mddamato commented on roles/rke2_common/tasks/cis-hardening.yml:
```diff
@@ -40,7 +40,7 @@
         mode: 0600
       register: sysctl_operation_tarball
-      when: ansible_facts['os_family'] != 'RedHat' or rke2_binary_tarball_check.stat.exists
+      when: ansible_facts['os_family'] != 'RedHat' or ansible_facts['os_family'] != 'Rocky' or rke2_binary_tarball_check.stat.exists
```
Should be an AND operator here. If evaluated for "Rocky" the when statement comes to true because it's !RedHat

FathiQ replied:
Fixed

mddamato commented on roles/rke2_common/tasks/main.yml:
```diff
@@ -19,7 +19,7 @@
   include: tarball_install.yml
   when:
-    - ansible_facts['os_family'] != 'RedHat' or rke2_binary_tarball_check.stat.exists == true
+    - ansible_facts['os_family'] != 'RedHat' or ansible_facts['os_family'] != 'Rocky' or rke2_binary_tarball_check.stat.exists == true
```
Same as the above comment https://github.com/rancherfederal/rke2-ansible/pull/64/files#r682706026

FathiQ replied:
Fixed too.

mddamato commented on roles/rke2_common/tasks/cis-hardening.yml:
```diff
@@ -40,7 +40,7 @@
         mode: 0600
       register: sysctl_operation_tarball
-      when: ansible_facts['os_family'] != 'RedHat' or rke2_binary_tarball_check.stat.exists
+      when: ansible_facts['os_family'] != 'RedHat' or ansible_facts['os_family'] != 'Rocky' and rke2_binary_tarball_check.stat.exists
```
I meant `(ansible_facts['os_family'] != 'RedHat' and ansible_facts['os_family'] != 'Rocky') or rke2_binary_tarball_check.stat.exists`

FathiQ replied:
oh my bad! done

mddamato commented on roles/rke2_common/tasks/cis-hardening.yml:
```diff
@@ -40,7 +40,7 @@
         mode: 0600
       register: sysctl_operation_tarball
-      when: ansible_facts['os_family'] != 'RedHat' or rke2_binary_tarball_check.stat.exists
+      when: ansible_facts['os_family'] != 'RedHat' and ansible_facts['os_family'] != 'Rocky' or rke2_binary_tarball_check.stat.exists
```
Are the parenthesis needed? we want ((A && B) || C) 

@dgvigil can you confirm?

dgvigil commented:
@FathiQ  When I run the playbook with Rocky 8.4, it works as intended without modifications.  What errors are you seeing with Rocky? Rocky is currently returning os_family of RedHat for me.

```
❯ ansible -i 143.198.8.129, all -u root -m setup -a 'filter=ansible_distribution,ansible_distribution_version,ansible_os_family'
143.198.8.129 | SUCCESS => {
    "ansible_facts": {
        "ansible_distribution": "Rocky",
        "ansible_distribution_version": "8.4",
        "ansible_os_family": "RedHat",
        "discovered_interpreter_python": "/usr/libexec/platform-python"
    },
    "changed": false
}
❯ ansible --version | head -1
ansible [core 2.11.2]
```
```
NAME        STATUS   ROLES                       AGE     VERSION          INTERNAL-IP     EXTERNAL-IP   OS-IMAGE                           KERNEL-VERSION                CONTAINER-RUNTIME
dvigil-01   Ready    <none>                      6m21s   v1.21.3+rke2r1   143.198.8.70    <none>        Rocky Linux 8.4 (Green Obsidian)   4.18.0-305.3.1.el8_4.x86_64   containerd://1.4.8-k3s1
dvigil-02   Ready    <none>                      6m21s   v1.21.3+rke2r1   143.198.8.88    <none>        Rocky Linux 8.4 (Green Obsidian)   4.18.0-305.3.1.el8_4.x86_64   containerd://1.4.8-k3s1
dvigil-03   Ready    control-plane,etcd,master   9m30s   v1.21.3+rke2r1   143.198.8.82    <none>        Rocky Linux 8.4 (Green Obsidian)   4.18.0-305.3.1.el8_4.x86_64   containerd://1.4.8-k3s1
dvigil-04   Ready    control-plane,etcd,master   10m     v1.21.3+rke2r1   143.198.8.129   <none>        Rocky Linux 8.4 (Green Obsidian)   4.18.0-305.3.1.el8_4.x86_64   containerd://1.4.8-k3s1
dvigil-05   Ready    control-plane,etcd,master   13m     v1.21.3+rke2r1   143.198.8.134   <none>        Rocky Linux 8.4 (Green Obsidian)   4.18.0-305.3.1.el8_4.x86_64   containerd://1.4.8-k3s1
```

mddamato commented:
Agree, I would think Rocky would work without modification but I can't find any documentation that specifies Rocky falling under RedHat specifically. I guess that test confirms?

FathiQ commented:
guys it's failed with me and stuck here in this step forever after my modification it's working fine 

`TASK [rke2_common : Reboot the machine (Wait for 5 min)] *********************************************************************************************************
changed: [172.30.201.181]
changed: [172.30.202.151]
changed: [172.30.203.158]
Monday 09 August 2021  16:07:28 +0300 (0:00:48.651)       0:02:08.426 ********* 
included: /Users/admin/rocky/rke2-ansible/roles/rke2_server/tasks/first_server.yml for 172.30.201.181
Monday 09 August 2021  16:07:28 +0300 (0:00:00.463)       0:02:08.890 ********* `

`[root@dev-rke2-server-01 ~]# cat /etc/redhat-release 
Rocky Linux release 8.4 (Green Obsidian)`

FathiQ commented:
@dgvigil @mddamato I think I found the issue, if you are using the AWS version you will get Rocky as os family.

`"ansible_distribution_file_path": "/etc/redhat-release", "ansible_distribution_file_variety": "RedHat", "ansible_distribution_file_parsed": true, "ansible_os_family": "Rocky"`

FathiQ commented on roles/rke2_common/tasks/main.yml:
```diff
@@ -19,7 +19,7 @@
   include: tarball_install.yml
   when:
-    - ansible_facts['os_family'] != 'RedHat' or rke2_binary_tarball_check.stat.exists == true
+    - (ansible_facts['os_family'] != 'RedHat' and ansible_facts['os_family'] != 'Rocky') or rke2_binary_tarball_check.stat.exists == true
```
@mddamato Done.

mddamato commented on roles/rke2_common/tasks/cis-hardening.yml:
```diff
@@ -40,7 +40,7 @@
         mode: 0600
       register: sysctl_operation_tarball
-      when: ansible_facts['os_family'] != 'RedHat' or rke2_binary_tarball_check.stat.exists
+      when: (ansible_facts['os_family'] != 'RedHat' and ansible_facts['os_family'] != 'Rocky') or rke2_binary_tarball_check.stat.exists
```
I tried to comment on this one too but it appears Github had some issues earlier today and commenting stopped working for a while right when I was trying to type. 
```
when: >-
  (ansible_facts['os_family'] != 'RedHat' and
  ansible_facts['os_family'] != 'Rocky') or
  rke2_binary_tarball_check.stat.exists
```
to pass linter

FathiQ replied:
Done

mddamato commented:
@FathiQ Thank you. There are still some trailing spaces but we'll fix those in the next PR. I appreciate your feedback and contribution.
